gpt4all-bindings/python/gpt4all/gpt4all.py

```diff
@@ -76,43 +73,38 @@ def retrieve_model(model_name: str, model_path: str = None, allow_download: bool
         """
         
         model_filename = model_name
-        if ".bin" not in model_filename:
+        if not model_filename.endswith(".bin"):
             model_filename += ".bin"
 
         # Validate download directory
-        if model_path == None:
+        if model_path is None:
+            try:
+                os.makedirs(DEFAULT_MODEL_DIRECTORY, exist_ok=True)
+            except OSError as exc:
+                raise ValueError(f"Failed to create model download directory at {DEFAULT_MODEL_DIRECTORY}: {exc}. "
+                                 "Please specify model_path.")
             model_path = DEFAULT_MODEL_DIRECTORY
-            if not os.path.exists(DEFAULT_MODEL_DIRECTORY):
-                try:
-                    os.makedirs(DEFAULT_MODEL_DIRECTORY)
-                except:
-                    raise ValueError("Failed to create model download directory at ~/.cache/gpt4all/. \
-                    Please specify download_dir.")
         else:
             model_path = model_path.replace("\\", "\\\\")
 
-        if os.path.exists(model_path):
-            model_dest = os.path.join(model_path, model_filename).replace("\\", "\\\\")
-            if os.path.exists(model_dest):
-                print("Found model file.")
-                return model_dest
-
-            # If model file does not exist, download
-            elif allow_download: 
-                # Make sure valid model filename before attempting download
-                model_match = False
-                for item in GPT4All.list_models():
-                    if model_filename == item["filename"]:
-                        model_match = True
-                        break
-                if not model_match:
-                    raise ValueError(f"Model filename not in model list: {model_filename}")
-                return GPT4All.download_model(model_filename, model_path)
-            else:
-                raise ValueError("Failed to retrieve model")
+        if not os.path.exists(model_path):
+            raise ValueError(f"Invalid model directory: {model_path}")
+
+        model_dest = os.path.join(model_path, model_filename).replace("\\", "\\\\")
+        if os.path.exists(model_dest):
+            print("Found model file at ", model_dest)
```

Comment:
Code looks great! Can't find the issue but there was a request to have this print statement be optional. Could you add a verbose keyword arg to this method and set default to True?

Reply:
Added verbosity to all methods in the file